lib/pxcors.js

```diff
@@ -0,0 +1,55 @@
+const { ORIGIN_HEADER, ACCESS_CONTROL_REQUEST_METHOD_HEADER } = require('./utils/constants');
+
+function isPreflightRequest(request) {
+    return request.method.toUpperCase() === 'OPTIONS' && request.get(ORIGIN_HEADER) && request.get(ACCESS_CONTROL_REQUEST_METHOD_HEADER);
+}
+
+function runPreflightCustomHandler(pxConfig, request) {
+    const corsCustomPreflightFunction = pxConfig.CORS_CUSTOM_PREFLIGHT_HANDLER;
+
+    if (corsCustomPreflightFunction) {
+        try {
+            return corsCustomPreflightFunction(request);
+        } catch (e) {
+            pxConfig.logger.debug(`Error while executing custom preflight handler: ${e}`);
+        }
+    }
+
+    return null;
+}
+
+function isCorsRequest(request, pxConfig) {
+    return request.get(ORIGIN_HEADER) && pxConfig.CORS_SUPPORT_ENABLED;
```

Comment:
I would remove the && pxConfig.CORS_SUPPORT_ENABLED as this is related to the feature from this function and check this outside of it.
if(isCorsRequest && pxConfig.CORS_SUPPORT_ENABLED)